client/network/src/lib.rs

```diff
@@ -222,6 +218,7 @@ impl TryFrom<&'_ CollationSecondedSignal> for BlockAnnounceData {
 /// chain. If it is at the tip, it is required to provide a justification or otherwise we reject
 /// it. However, if the announcement is for a block below the tip the announcement is accepted
 /// as it probably comes from a node that is currently syncing the chain.
+#[derive(Clone)]
```

Comment:
For what do we need the clone?

Reply:
I was having lifetime issues while moving function calls to the async block in the validate method: https://github.com/paritytech/cumulus/pull/909/files#diff-0fe3ce0d7087ba7f979d9065d8d812b23562d2f4ed5341719aa8480dc48c74cdR345

Reply:
Done